src/bundles/files.js

```diff
@@ -119,26 +119,31 @@ function readAsBuffer (file) {
   })
 }
 
-bundle.doFilesWrite = (root, files) => ({dispatch, getIpfs, store}) => {
+bundle.doFilesWrite = (root, files) => async ({dispatch, getIpfs, store}) => {
   dispatch({ type: 'FILES_WRITE_STARTED' })
 
-  let promises = []
+  try {
+    let promises = []
 
-  for (const file of files) {
-    promises.push(readAsBuffer(file))
-  }
+    for (const file of files) {
+      promises.push(readAsBuffer(file))
+    }
+
+    files = await Promise.all(promises)
+
+    for (const file of files) {
+      await getIpfs().files.mkdir(join(root, dirname(file.name)), { parents: true })
+    }
 
-  return Promise.all(promises).then(files => {
-    return Promise.all(files.map((file) => {
+    await Promise.all(files.map(file => {
       const target = join(root, file.name)
       return getIpfs().files.write(target, file.content, { create: true })
```

Comment:
We should  avoid `files.write`, as it will produce different CID than `add`.
See details and workaround in: https://github.com/ipfs-shipyard/ipfs-webui/issues/676